pkg/controllers/sync/controller.go

```diff
@@ -430,10 +445,77 @@ func (s *SyncController) syncToClusters(ctx context.Context, fedResource Federat
 			string(fedcorev1a1.ClusterRetrievalFailed),
 			errors.Wrap(err, "Failed to retrieve list of clusters"),
 		)
-		return s.setFederatedStatus(ctx, fedResource, fedcorev1a1.ClusterRetrievalFailed, nil)
+		return nil, nil, s.setFederatedStatus(ctx, fedResource, fedcorev1a1.ClusterRetrievalFailed, nil)
+	}
+	joined := make(map[string]*fedcorev1a1.FederatedCluster, len(clusters))
+	for _, cluster := range clusters {
+		joined[cluster.Name] = cluster
 	}
 
 	selectedClusterNames := fedResource.ComputePlacement(clusters)
+	pendingCreateClusters := selectedClusterNames.Clone()
+	status := fedResource.Object().GetStatus()
+	for _, cluster := range status.Clusters {
+		pendingCreateClusters.Delete(cluster.Cluster)
+		if cluster, exist := joined[cluster.Cluster]; exist {
+			requireSync = append(requireSync, cluster)
+		}
+	}
+
+	if pendingCreateClusters.Len() == 0 {
+		return requireSync, selectedClusterNames, worker.StatusAllOK
+	}
+	for cluster := range pendingCreateClusters {
+		if cluster, exist := joined[cluster]; exist {
```

Comment:
We should also account for terminating clusters and cascading delete here, to prevent an unnecessary and potentially misleading status update, since we do not sync to terminating clusters.